Gems/ROS2/Code/Source/Gripper/FingerGripperComponent.cpp

```diff
@@ -0,0 +1,288 @@
+                    ->Attribute(AZ::Edit::Attributes::Category, "ROS2")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &FingerGripperComponent::m_velocityEpsilon,
+                        "Velocity Epsilon",
+                        "The maximum velocity to consider the gripper as stalled.")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &FingerGripperComponent::m_distanceEpsilon,
+                        "Distance Epsilon",
+                        "The maximum distance to gripper can be from the goal to have reached it.")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &FingerGripperComponent::m_stallTime,
+                        "Stall Time",
+                        "The time to wait before considering the gripper as stalled.");
+            }
+        }
+    }
+
+    AZStd::string GetJointName(AZ::EntityId entityId)
+    {
+        AZ::Entity* entity{ nullptr };
+        AZ::ComponentApplicationBus::BroadcastResult(entity, &AZ::ComponentApplicationRequests::FindEntity, entityId);
+        AZ_Assert(entity, "Entity %s not found.", entityId.ToString().c_str());
+        if (entity)
+        {
+            ROS2FrameComponent* component = entity->FindComponent<ROS2FrameComponent>();
+            if (component)
+            {
+                return component->GetJointName().GetStringView();
+            }
+        }
+        return AZStd::string();
+    }
+
+    ManipulationJoints& FingerGripperComponent::GetFingerJoints()
+    {
+        m_fingerJoints.clear();
+        m_rootOfArticulation = Utils::GetRootOfArticulation(GetEntityId());
+        AZ_Warning(
+            "FingerGripperComponent",
+            m_rootOfArticulation.IsValid(),
+            "Entity %s is not part of an articulation.",
+            GetEntity()->GetName().c_str());
+        ManipulationJoints allJoints;
+        if (m_rootOfArticulation.IsValid())
+        {
+            JointsManipulationRequestBus::EventResult(allJoints, m_rootOfArticulation, &JointsManipulationRequests::GetJoints);
+        }
+        AZStd::vector<AZ::EntityId> descendantIds;
+        AZ::TransformBus::EventResult(descendantIds, GetEntityId(), &AZ::TransformBus::Events::GetAllDescendants);
+
+        for (AZ::EntityId descendant : descendantIds)
+        {
+            AZStd::string jointName = GetJointName(descendant);
+            if (!jointName.empty())
+            {
+                AZ_Printf("FingerGripperComponent", "Adding finger joint %s", jointName.c_str());
+                m_fingerJoints[jointName] = allJoints[jointName];
+            }
+        }
+
+        return m_fingerJoints;
+    }
+
+    void FingerGripperComponent::SetPosition(float position, float maxEffort)
+    {
+        // A check to make this work with moveit
```

Comment:
Suggestion: move comment inside the if (in the line with the opening bracket). Change the text to explanation what moveIt uses infinity for